packages/core/src/routes/interaction/utils/interaction.ts

```diff
@@ -128,33 +128,58 @@ export const clearInteractionStorage = async (ctx: Context, provider: Provider)
 export const assignConnectorSessionResult = async (
   ctx: Context,
   provider: Provider,
-  connectorSession: ConnectorSession
+  connectorSession: ConnectorSession,
+  mode: 'replace' | 'merge' = 'replace'
 ) => {
-  const details = await provider.interactionDetails(ctx.req, ctx.res);
+  const { result } = await provider.interactionDetails(ctx.req, ctx.res);
+
+  const connectorSessionResult = z
+    .object({
+      connectorSession: connectorSessionGuard,
+    })
+    .catchall(z.unknown())
+    .safeParse(result);
+
+  if (result && connectorSessionResult.success && mode === 'merge') {
+    const { connectorSession: originalConnectorSession, ...rest } = connectorSessionResult.data;
+    await provider.interactionResult(ctx.req, ctx.res, {
+      ...rest,
+      connectorSession: { ...originalConnectorSession, ...connectorSession },
+    });
+
+    return;
+  }
+
   await provider.interactionResult(ctx.req, ctx.res, {
-    ...details.result,
+    ...result,
     connectorSession,
   });
 };
 
 export const getConnectorSessionResult = async (
   ctx: Context,
-  provider: Provider
+  provider: Provider,
+  clearAfterGet = true
```

Comment:
When declaring a parameter, we usually avoid using `true` as default value. Consider reversing the logic